src/language/__tests__/visitor-test.js

```diff
@@ -95,6 +94,34 @@ describe('Visitor', () => {
     ]);
   });
 
+  it('validates ancestors argument', () => {
+    const ast = parse('{ a }', { noLocation: true });
+    const nodesInPath = [];
+
+    visit(ast, {
+      enter(node, key, parent, path, ancestors) {
+        const inArray = typeof key === 'number';
+        const expectedAncestors = nodesInPath.slice(0, path.length - 1);
+        expect(ancestors).to.deep.equal(expectedAncestors);
```

Comment:
Why is this not `expect(ancestors).to.deep.equal(nodesInPath);`? i.e. why is the slicing needed?

Reply:
@mjmahone Because we need `nodesInPath` without last element and `slice(0, path.length - 1)` does exactly that it returns an array without last element.

Basically, it tests this statement:
> All nodes and Arrays visited before reaching parent of this node

Reply:
@mjmahone I rewrote this test to make it more explicit.